src/help/help_topic_generators.cpp

```diff
@@ -650,35 +671,31 @@ std::string unit_topic_generator::operator()() const {
 			}
 
 			if (info.union_type().size() == 1 && info.union_type()[0] == info.number() && info.is_nonnull()) {
-				terrain_moves.push_back(std::make_tuple(info.name(), // 0 name (t_string)
+				terrain_movement_info movement_info = 
+				{
+					info.name(), // 0 name (t_string)
 					info.id(),  // 1 id (t_string)
 					100 - movement_type.defense_modifier(terrain), // 2 defense (int)
 					moves, // 3 movement_cost (int)
 					movement_type.vision_cost(terrain), // 4 vision_cost (int)
 					movement_type.jamming_cost(terrain), // 5 jamming_cost (int)
-					movement_type.get_defense().capped(terrain))); // 6 defense_cap (bool)
+					movement_type.get_defense().capped(terrain) // 6 defense_cap (bool)
+				};
+
+				terrain_moves.insert(movement_info);
 			}
 		}
 
-		// SORT HERE
-		auto movement_compare_ascending = [](std::tuple<t_string, t_string, int, int, int, int, bool> a,
-			std::tuple<t_string, t_string, int, int, int, int, bool> b)
-		{
-			return std::get<0>(a) < std::get<0>(b);
-		};
-
-		std::sort(terrain_moves.begin(), terrain_moves.end(), movement_compare_ascending); 
-
-		for(unsigned int i = 0; i < terrain_moves.size(); i++)
+		for(const terrain_movement_info &terr_move : terrain_moves)
 		{
 			std::vector<item> row;
-			const std::string& name = std::get<0>(terrain_moves[i]); // get name
-			const std::string& id = std::get<1>(terrain_moves[i]); // get id
-			const int defense = std::get<2>(terrain_moves[i]); // get defense
-			const int moves = std::get<3>(terrain_moves[i]); // get moves
-			const int views = std::get<4>(terrain_moves[i]); // get vision cost
-			const int jams  = std::get<5>(terrain_moves[i]); // get jamming cost
-			const bool has_cap = std::get<6>(terrain_moves[i]); // has defense cap
+			const std::string& name = terr_move.name; // get name
```

Comment:
You don't need to assign the values to local variables, you can just access them via the struct.

And the comments about which variable you get are definitely redundant.